src/Storage/Session.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace PhpAb\Storage;
+
+use InvalidArgumentException;
+
+/**
+ * Stores the participation state of the user in a session.
+ */
+class Session implements StorageInterface
+{
+    /**
+     * @var string The namespace of the session.
+     */
+    private $namespace;
+
+    /**
+     * Initializes a new instance of this class.
+     *
+     * @param string $namespace The namespace of the session.
+     */
+    public function __construct($namespace)
+    {
+        if (!$namespace) {
```

Comment:
```0.3``` is no valid namespace, too. check for ```is_string``` and leave a comment that we can typehint that if php5.x is abandoned in some years